src/rules/core/fields-not-in-model-rule.js

```diff
@@ -251,7 +256,269 @@ module.exports = class FieldsNotInModelRule extends Rule {
+                if (typeof subContext === 'string') {
+                  const subJsonResponse = await JsonLoaderHelper.getFileAsync(subContext, node.options);
+                  const subReturnedContext = subJsonResponse.data;
+                  if (
+                    typeof subReturnedContext['@context'] === 'object'
+                    && subReturnedContext['@context'] !== null
+                    && !(subReturnedContext['@context'] instanceof Array)
+                  ) {
+                    returnedContext['@context'][index] = subReturnedContext['@context'];
+                  } else {
+                    errors.push(
+                      this.createError(
+                        'extensionComplex',
+                        {
+                          value: subContext,
+                          path: node.getPath(field),
+                        },
+                        {
+                          context: subContext,
+                        },
+                      ),
+                    );
+                  }
+                  if (typeof subReturnedContext['@graph'] !== 'undefined') {
+                    errors.push(
+                      this.createError(
+                        'extensionGraph',
+                        {
+                          value: subContext,
+                          path: node.getPath(field),
+                        },
+                        {
+                          context: subContext,
+                        },
+                      ),
+                    );
+                  }
+                }
+                index += 1;
+              }
+            }
+          }
+          returnContexts.push(returnedContext);
+        } else if (
+          jsonResponse.statusCode !== 200
+          && jsonResponse.statusCode !== null
+        ) {
+          errors.push(
+            this.createError(
+              'extensionErrorCode',
+              {
+                value: url,
+                path: node.getPath(field),
+              },
+              {
+                context: url,
+                code: jsonResponse.statusCode,
+                field,
+              },
+            ),
+          );
+        } else {
+          errors.push(
+            this.createError(
+              'extensionInvalid',
+              {
+                value: url,
+                path: node.getPath(field),
+              },
+              {
+                context: url,
+                field,
+              },
+            ),
+          );
+        }
+      } else if (typeof url === 'object' && url !== null) {
+        returnContexts.push(url);
+      }
+    }
+    return {
+      contexts: returnContexts,
+      errors,
+    };
+  }
+
+  /**
+   * @deprecated since version 1.2.0, since it uses synchronous IO
+   *   (https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/#comparing-code). Use
+   *   validateFieldAsync() instead
+   */
+  validateFieldSync(node, field) {
```

Comment:
This method is the exact same as the previous `validateField` method - Git is just marking it as new as it's been moved around